### PR TITLE
Remove deprecated code paths that trigger a runtime notice

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/BrowserKit/AbstractBrowser.php [hunk 1]
```diff
@@ -450,9 +450,9 @@ protected function doRequestInProcess($request)
             unlink($deprecationsFile);
             foreach ($deprecations ? unserialize($deprecations) : [] as $deprecation) {
                 if ($deprecation[0]) {
-                    @trigger_error($deprecation[1], E_USER_DEPRECATED);
+                    @trigger_error($deprecation[1], E_USER_DEPRECATED); //??XXX
                 } else {
-                    @trigger_error($deprecation[1], E_USER_DEPRECATED);
+                    @trigger_error($deprecation[1], E_USER_DEPRECATED); //??XXX
```
Comment (nicolas-grekas) [inline src/Symfony/Component/BrowserKit/AbstractBrowser.php]: looks like there us a bug here? anyone would like to have a look?

File: src/Symfony/Component/Security/Http/Session/SessionAuthenticationStrategy.php [hunk 2]
```diff
@@ -48,7 +48,7 @@ public function onAuthentication(Request $request, TokenInterface $token)
 
             case self::MIGRATE:
                 // Note: this logic is duplicated in several authentication listeners
-                // until Symfony 5.0 due to a security fix with BC compat
+                // until Symfony 5.0 due to a security fix with BC compat XXX
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Http/Session/SessionAuthenticationStrategy.php]: What does this mean for 5.0 @weaverryan? What should be done now?

Comment (Simperfit) [reply src/Symfony/Component/Security/Http/Session/SessionAuthenticationStrategy.php]: removing the duplicated logic in other listener I think

**Comment by lyrixx** [comment]
:+1: for the workflow part

File: src/Symfony/Component/BrowserKit/AbstractBrowser.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/BrowserKit/AbstractBrowser.php]: Found why, see #31962

File: src/Symfony/Component/Security/Http/Session/SessionAuthenticationStrategy.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Security/Http/Session/SessionAuthenticationStrategy.php]: this is the only remaining part that I didn't update in this PR
@weaverryan maybe, up for a PR?

**Pull request merged**